content/sklearn/concepts/clustering/clustering.md

```diff
@@ -0,0 +1,105 @@
+
+Visualling look at the clusters, Silhouette Score, rand index and homogeneity are among the few methods to evaluate the clusters.
+
+## Syntax
+Sklearn provides the `KMeans` class for implementing clustering.
+
+```psuedo
+KMeans(n_clusters: int = 8, *, init: str = "k-means++", n_init: str = "auto", max_iter: int = 300, tol: float = 0.0001, verbose: int = 0, random_state: Any | None = None, copy_x: bool = True, algorithm: str = "lloyd") 
+```
+
+`KMeans` has the following parameters:
+
+- `n_clusters` (int, default=8): The number of clusters.
+- `init` (array, default=k-means++): It chooses the method of initialization. If `k-means++`, then a random point is choosen for the initial centroid and for the remaining centroids each data point is weighed according to the distance squared from the initial centroid. If `random`, then random points are choosen for the initial centroid. If `array`, then specific set of points are used as initial centroids.
+- `n_init` (int, default=auto): The number of times the algorithm will run with different initial centroid positions. If `1`, the algorithm runs only once with one random initialization of centroids.
+- `max_iter` (int, default=300): Maximum number of iterations of the KMeans algorithm for a single run.
+- `tol` (float, default:1e-4): The tolerance for stopping the algorithm when the change in centroids is smaller than this value between iterations.
+- `verbose` (int, default=0): Controls the amount of output information printed during the K-Means fitting process, useful for debugging. Higher values show more details.
+- `random_state` (int, default=None): Determines random number generation for centroid initialization.
+- `copy_x` (bool, default=True): If `True`, then original data is copied before processing and if `false`, then it modifies the input data by subtracting and then adding the data mean, can save memory.
+- `algorithm` (default=lloyd): If `lloyd`, then the traditional algorithm is used, and if `elkan`, then triangle inequality is used on datasets with well-defined 
+clusters. 
+
+## Example
+
+This example uses the Iris dataset to form the clusters.
+
+```py
+#import all the necessary libraries
+import pandas as pd
+import numpy as np
+from sklearn import datasets
+from sklearn.cluster import KMeans
+
+# Dataset
+df= datasets.load_iris()
+d = pd.DataFrame(df.data, columns=df.feature_names)
+print(d.head)
+print(d.columns)
+
+#Initializing the k-means model
+model=KMeans(n_clusters=3,n_init=1)
+y=model.fit(d)
+
+# Get the cluster labels
+y = model.labels_
+
+# Print the cluster labels
+print(y)
+
+# Get the unique clusters
+clusters = pd.unique(y)
+
+# Print the unique clusters
+print(clusters)
+```
+
+If you want to visually access the clusters:
+
+```py
+from sklearn.decomposition import PCA
+import matplotlib.pyplot as plt
+
+pca = PCA(n_components=2)
+d_pca = pca.fit_transform(d)
+
+plt.scatter(d_pca[:, 0], d_pca[:, 1], c=y)
+plt.title("K-Means Clustering")
+plt.show()
+```
```

Comment:
Please add the output image for this as well. You can place the .png file in the "media" folder in the docs